src/domain/app/AppYouthProfileRoute.tsx

```diff
@@ -23,21 +22,19 @@ function AppYouthProfileRoute(props: Props) {
   const { t } = useTranslation();
   const [showNotification, setShowNotification] = useState<boolean>(false);
 
-  const [loadProfile, { data, loading }] = useLazyQuery<HasYouthProfile>(
-    HAS_YOUTH_PROFILE,
-    {
-      onError: (error: Error) => {
-        Sentry.captureException(error);
-        setShowNotification(true);
-      },
-    }
-  );
-
-  const isAuthenticated = useSelector(isAuthenticatedSelector);
-
-  useEffect(() => {
-    if (isAuthenticated) loadProfile();
-  }, [isAuthenticated, loadProfile]);
+  const { data, loading } = useQuery<HasYouthProfile>(HAS_YOUTH_PROFILE, {
+    onError: error => {
+      // Ignore permission errors. This decision was made based on the difficulty to add enough checks that slow down Redirecting.
+      // Using useLazyQuery with isAuthenticatedSelector won't fetch profile information fast enough and user is always
+      // redirected to LoginPage. If in the future permission error causes application to crash, some kind of custom hook is necessary.
+      error.graphQLErrors.forEach(qlError => {
+        if (qlError.message !== authConstants.PERMISSION_DENIED) {
+          Sentry.captureException(qlError);
```

Comment:
You are passing the error to Sentry within a loop. A lot of the time we only pass the error object once

```tsx
onError: (error: Error) => {
  Sentry.captureException(error);
  // ...rest
}
```

I don't actually know which approach is more correct, because I don't have an idea of how Sentry behaves in the two cases. I just want to point this out.

Reply:
If I pass down it like that, permission errors are sent as well and thats what I'm trying to prevent. I find it kind of impossible that there would be more than one error (now that permission is excluded that is).

Reply:
This is a pattern I used in another project:

```tsx
onError: (error: Error) => {
  if (!includesError(error.graphQLErrors, permissionError)) {
    Sentry.captureException(error);
    // ...rest
  }
}
```

But I don't think that you should spend time on changing this. If I needed to guess, I'd say that your approach is the more correct one. Maybe I wrote this down, because it's more of a mental note for me that this is something I should keep in mind when implementing error handling in the future.